test/test_util_pipe.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2022 skelly37
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+import concurrent.futures
+
+from test.picardtestcase import PicardTestCase
+
+from picard import (
+    PICARD_APP_NAME,
+    PICARD_FANCY_VERSION_STR,
+)
+from picard.util import pipe
+
+
+def pipe_listener(pipe_handler, end_of_sequence):
+    received = []
+    messages = []
+    while end_of_sequence not in messages:
+        messages = pipe_handler.read_from_pipe()
+        for message in messages:
+            if message not in (pipe.Pipe.MESSAGE_TO_IGNORE, pipe.Pipe.NO_RESPONSE_MESSAGE, "", end_of_sequence):
```

Comment:
Since this sequence doesn't change, it would be better to define it as a set outside those loops.